### PR TITLE
[pkg/otlp] Deep copy values when reading a configuration section

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Deep copy objects gotten via the `Get` method 

### Motivation

Viper returns a reference to maps/slices, which can cause concurrency issues. This avoids concurrent modifications of the returned references.

### Additional Notes

We could also deep copy at the `pkg/config` level, but I am not sure what the implications of that would be.

### Describe how to test/QA your changes

See https://github.com/DataDog/datadog-agent/pull/12985#issuecomment-1204948693

<To be filled>

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by mx-psi** [comment]
cc @AlexandreYang to provide instructions for reproducing the issue

File: releasenotes/notes/fix-otlp-concurrency-4966bff7ebac155e.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Fix concurrent map access issues when using OTLP ingest
```
Comment (buraizu) [inline releasenotes/notes/fix-otlp-concurrency-4966bff7ebac155e.yaml:11]:
```suggestion
    Fix concurrent map access issues when using OTLP ingest.
```

**Review by buraizu — CHANGES_REQUESTED** [review]

**Comment by AlexandreYang** [comment]
Hi @mx-psi , the issue occurred when running normally `./bin/agent/agent run -c dev/dist/datadog.yaml`.

I had few snmp integrations instances, that might have contributed to make it more prominent:

```
init_config:
  loader: core
  use_device_id_as_hostname: true  # recommended

instances:
- community_string: aos
  network_address: "127.0.0.1/32"
  port: 1161

- community_string: aos6
  network_address: "127.0.0.2/32"
  port: 1161

- community_string: ciena-sds
  network_address: "127.0.0.3/32"
  port: 1161
```

The panic happen quite rarely though, maybe once every 20-30 runs. So, making it harder to spot in which case it happen.

**Comment by mx-psi** [comment]
@AlexandreYang Thanks, that's helpful!

**Review by KSerrania — APPROVED** [review]

**Review by buraizu — APPROVED** [review]

**Pull request merged**